tests/unit/core/scan/test_commit.py

```diff
@@ -392,6 +388,30 @@ def test_from_sha(
     ]
 
 
+def test_from_sha_files_matches_content(tmp_path):
+    """
+    GIVEN a commit with many files
+    WHEN Commit.get_files() is called
+    THEN the reported file names match their expected content
+    """
+    repo = Repository.create(tmp_path)
+
+    for idx in range(50):
+        path = tmp_path / str(idx)
+        path.parent.mkdir(exist_ok=True)
+        path.write_text(f"{idx}\n")
+        repo.add(path)
+    repo.create_commit()
+
+    sha = repo.get_top_sha()
+    commit = Commit.from_sha(sha, cwd=tmp_path)
+    files = list(commit.get_files())
+
+    for file in files:
+        last_line = file.content.splitlines()[-1]
+        assert last_line == f"+{file.path.name}"
+
+
```

Comment:
Do you think it would be doable to add a test that in the case of a merge commit we take into account the change from both parents ?

Reply:
I think `scenario_merge_with_changes` does this, but maybe I misunderstand what you are suggesting?

Reply:
I just added a test to inspect the content of what we scan in case of a merge commit with conflicts.